doc/sphinx/proof-engine/ltac.rst

```diff
@@ -28,14 +29,27 @@ especially about its foundations, please refer to :cite:`Del00`.
 Syntax
 ------
 
-The syntax of the tactic language is given below. See Chapter
-:ref:`gallinaspecificationlanguage` for a description of the BNF metasyntax used
-in these grammar rules. Various already defined entries will be used in this
-chapter: entries :token:`num`, :token:`int`, :token:`ident`
-:token:`qualid`, :token:`term`, :token:`cpattern` and :token:`tactic`
-represent respectively natural and integer numbers,
-identifiers, qualified names, Coq terms, patterns and the atomic
-tactics described in Chapter :ref:`tactics`.
+The syntax of the tactic language is given below.
+
+The main entry of the grammar is :n:`@ltac_expr`, which is used in proof mode
+as well as to define new tactics with the :cmd:`Ltac` command.
+
+The grammar uses multiple :n:`ltac_expr*` nonterminals to express how subexpressions
+are grouped when they're not fully parenthesized.  For example, in many programming
+languages, `a*b+c` is interpreted as `(a*b)+c` because `*` has
+higher precedence than `+`.  Usually `a/b/c` is given the :gdef:`left associative`
+interpretation `(a/b)/c` rather than the :gdef:`right associative` interpretation
+`a/(b/c)`.
+
+In Coq, the expression :n:`try repeat @tactic__1 || @tactic__2; @tactic__3; @tactic__4`
+is interpreted as :n:`(try (repeat (@tactic__1 || @tactic__2)); @tactic__3); @tactic__4`
+because `||` is part of :token:`ltac_expr2`, which has higher precedence than
+:tacn:`try` and :tacn:`repeat` (at the level of :token:`ltac_expr3`), which
+in turn have higher precedence than `;`, which is :term:`left associative` and part
```

Comment:
I'm a bit confused.  Semantically, at least, `;` is fully associative.  What's the significance of it being left-associative here?

Reply:
`;` is _parsed_ as left-associative, so a;b;c is (a;b);c.  But, yeah, a bit confusing because the underlying operation is associative.  Are there any operators where (a OP b) OP c <> a OP (b OP c)?  How about `+` or `||`?

Reply:
No, + and || are both associative as well.  The `; [ ... ]` construct is not associative with `;`, as `(a; b); [ c ]` is different from `a; (b; [ c ])`.  I don't know of any binary infix operations of Ltac that are not associative, though there may well be ones I can't recall.